### PR TITLE
Implement wlr_region using pixman

### PR DESCRIPTION
**Pull request by nyorain**
Rather small pr, makes wlr dependent on pixman.
Will be needed to further implement wl_surface.
Not sure if it is ok to not use something wlr_region and instead directly setting the pixman region as the regions resource user data.

### CODE REVIEW

File: include/wlr/types/wlr_region.h [hunk 1]
```diff
@@ -0,0 +1,10 @@
+#ifndef _WLR_TYPES_REGION_H
+#define _WLR_TYPES_REGION_H
+
+struct wl_resource;
+
+// Implements the given resource as region.
+// Sets the associated pixman_region32_t as userdata.
+void wlr_region_create(struct wl_resource *res);
```
Comment (SirCmpwn) [inline include/wlr/types/wlr_region.h]: Can you return the `pixman_region32_t`?

Comment (nyorain) [inline include/wlr/types/wlr_region.h]: thought about, are you sure this is a good idea? the region is directly connected with the resource so why should it be needed on creation? changing it independently from the client will result in weird behavior and it should only be read when it is passed by the client as parameter somewhere i guess

Comment (SirCmpwn) [inline include/wlr/types/wlr_region.h]: In case the user wants to stow it away somewhere to access it without relying on the user data.

Comment (nyorain) [inline include/wlr/types/wlr_region.h]: But it cannot know when the region is destroyed. The region is probably only created by the compositor implementation.

Comment (SirCmpwn) [inline include/wlr/types/wlr_region.h]: Alright, w/e.

**Pull request merged**